File: data/cpp/src/function_cvxpy_variable.cpp
#include <string>
#include <Eigen/Dense>
#include <sstream>
#include <stdexcept>
class CVXPYVariableFunction {
public:
    std::string process_variables(const std::string& x_value = "", 
                                const std::string& y_value = "", 
                                const std::string& constraint = ""){
    
                                    try {
                                        // 处理未赋值变量的情况
                                        if (x_value.empty() && y_value.empty() && constraint.empty()) {
                                            // 使用Eigen创建变量
                                            Eigen::MatrixXd x = Eigen::MatrixXd::Zero(1,1);
                                            Eigen::MatrixXd y = Eigen::MatrixXd::Zero(1,1);
                                            return "x, y";
                                        }
                                
                                        // 处理x和y值的情况
                                        if (!x_value.empty() && !y_value.empty()) {
                                            try {
                                                double x = std::stod(x_value);
                                                double y = std::stod(y_value);
                                                
                                                // 检查负值
                                                if (x < 0 || y < 0) {
                                                    return "Error: Variable size cannot be negative";
                                                }
                                                
                                                // 格式化输出
                                                std::ostringstream oss;
                                                oss << "x: " << x << ".0, y: " << y << ".0";
                                                return oss.str();
                                            }
                                            catch (const std::exception&) {
                                                return "Error: Invalid value for x or y";
                                            }
                                        }
                                
                                        // 处理约束条件
                                        if (!constraint.empty()) {
                                            return "Error: Invalid variable constraint";
                                        }
                                
                                        return "Error: Invalid input";
                                    }
                                    catch (const std::exception&) {
                                        return "Error: Unknown error occurred";
                                    }
                                }
};

File: data/python/src/function_cvxpy_variable.py
import cvxpy as cp

class CVXPYVariableFunction:
    def process_variables(self, x_value: str = None, y_value: str = None, constraint: str = None) -> str:
        if x_value is None and y_value is None and constraint is None:
           
            x = cp.Variable()
            y = cp.Variable()
            return str(x), str(y)

        if x_value is not None and y_value is not None:
            
            try:
                x = float(x_value)
                y = float(y_value)
                if x < 0 or y < 0:
                    return "Error: Variable size cannot be negative"
                return f"x: {x}, y: {y}"
            except ValueError:
                return "Error: Invalid value for x or y"

        if constraint is not None:
            if isinstance(constraint, str):
                return "Error: Invalid variable constraint"
            return "Constraint is valid"

        return "Error: Invalid input"
